### ISSUE TITLE
Support Selecting, Filtering and Expanding on Derived Entity Type Props

### ISSUE THREAD

**Issue by texttechne** [odata2ts/odata2ts#329]
Examples:

```
### cast to subtype and filter thereby
GET {{baseUrl}}/People('russellwhyte')/Trips(0)/PlanItems/Microsoft.OData.SampleService.Models.TripPin.Flight

### select & expand prop of derived type
GET {{baseUrl}}/People('russellwhyte')/Trips(0)?$expand=PlanItems($expand=Microsoft.OData.SampleService.Models.TripPin.Flight/Airline;$select=Microsoft.OData.SampleService.Models.TripPin.Flight/Airline)

### filter on casted prop of subtype
GET {{baseUrl}}/People('russellwhyte')/Trips(0)/PlanItems?$filter=Microsoft.OData.SampleService.Models.TripPin.Flight/FlightNumber eq 'AA26'
```
Only works with this Trippin service: `https://services.odata.org/V4/SESSION_ID/TripPinServiceRW`

**Comment by texttechne** [comment]
See [discussion](https://github.com/odata2ts/odata2ts/discussions/320)

**Comment by texttechne** [comment]
### Provided Solution
```ts
mainService.people("russellwhyte").trip(0).query((builder, qTrip) => 
  builder.expanding("planItems", (piBuilder, qPi) => 
    piBuilder.select("QFlight_flightNumber", "QFlight_Airline").expand("QFlight_Airline") 
  )
);
// result url:  /People('russellwhyte')/Trips(0)?$expand=PlanItems($expand=Trippin.Flight/Airline;$select=Trippin.Fligh/FlightNumber,Trppin.Flight/Airline)


mainService.people("russellwhyte").trip(0).planItems().query((piBuilder, qPi) => 
  piBuilder.filter(qPi.QFlight_flightNumber.eq("F123"))
)
// result url:  /People('russellwhyte')/Trips(0)/PlanItems?$filter=Trippin.Flight/FlightNumber eq 'F123'

```

**Comment by texttechne** [comment]
Released with [@odata2ts/odata2ts@0.39.0](https://github.com/odata2ts/odata2ts/releases/tag/%40odata2ts%2Fodata2ts%400.39.0)
